oauth2client/django_util/__init__.py

```diff
@@ -0,0 +1,284 @@
+
+   @oauth_enabled
+   def optional_oauth2(request):
+       if request.oauth.has_credentials():
+           # this could be passed into a view
+           # request.oauth.http is also initialized
+           return HttpResponse("User email: %s" % request.oauth.credentials.id_token['email'])
+       else:
+           return HttpResponse('Here is an OAuth Authorize link: <a href="%s">Authorize</a>' %
+                               request.oauth.get_authorize_redirect())
+
+If a view needs a scope not included in the default scopes specified in
+the settings, you can use [incremental auth](https://developers.google.com/identity/sign-in/web/incremental-auth)
+and specify additional scopes in the decorator arguments.
+
+.. code-block:: python
+   :caption: views.py
+   :name: views_required_additional_scopes
+
+   @oauth_enabled(scopes=['https://www.googleapis.com/auth/drive'])
+   def drive_required(request):
+       if request.oauth.has_credentials():
+           service = build(serviceName='drive', version='v2', http=request.oauth.http,
+                developerKey=API_KEY)
+           events = service.files().list().execute()['items']
+           return HttpResponse(str(events))
+       else:
+           return HttpResponse('Here is an OAuth Authorize link: <a href="%s">Authorize</a>' %
+                           request.oauth.get_authorize_redirect())
+
+
+To provide a callback on authorization being completed, use the oauth2_authorized
+signal:
+
+.. code-block:: python
+   :caption: views.py
+   :name: signals
+
+   from oauth2client.django_util.signals import oauth2_authorized
+
+   def test_callback(sender, request, credentials, **kwargs):
+       print "Authorization Signal Received %s" % credentials.id_token['email']
+
+   oauth2_authorized.connect(test_callback)
+
+"""
+import httplib2
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.urlresolvers import reverse
+from oauth2client import clientsecrets
+
+from .storage import get_storage
+
+GOOGLE_OAUTH2_DEFAULT_SCOPES = ('email', )
+GOOGLE_OAUTH2_REQUEST_ATTRIBUTE = 'oauth'
+
+
+def _load_client_secrets(filename):
+    """Loads client secrets from the given filename."""
+    client_type, client_info = clientsecrets.loadfile(filename)
+
+    if client_type != clientsecrets.TYPE_WEB:
+        raise ValueError(
+            'The flow specified in {} is not supported, only the WEB flow type is '
+            'supported.'.format(client_type))
+    return client_info['client_id'], client_info['client_secret']
+
+
+def _get_oauth2_client_id_and_secret(settings_instance):
+    """Initializes client id and client secret based on the settings"""
+    if getattr(settings, 'GOOGLE_OAUTH2_CLIENT_SECRETS_JSON', None):
+        return _load_client_secrets(
+            settings_instance.GOOGLE_OAUTH2_CLIENT_SECRETS_JSON)
+    elif getattr(settings_instance, "GOOGLE_OAUTH2_CLIENT_ID", None) and \
```

Comment:
"Do not use backslash line continuation.": https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length.